diff -Naur c9upstream/plugins/c9.ide.layout.classic/layout.js c9fork/plugins/c9.ide.layout.classic/layout.js
--- c9upstream/plugins/c9.ide.layout.classic/layout.js	2017-04-09 19:19:24.599925145 +0200
+++ c9fork/plugins/c9.ide.layout.classic/layout.js	2017-04-11 10:44:25.359064655 +0200
@@ -269,6 +269,8 @@
             }
             if (obj.name == "save") 
                 return plugin.getElement("barTools");
+            if (obj.name == "zamger.gui") 
+                return plugin.getElement("barTools");
             if (obj.name == "run.gui") 
                 return plugin.getElement("barTools");
             else if (obj.name == "console") {
