diff -Naur c9upstream/plugins/node_modules/ace/lib/ace/snippets/c_cpp.snippets c9fork/plugins/node_modules/ace/lib/ace/snippets/c_cpp.snippets
--- c9upstream/plugins/node_modules/ace/lib/ace/snippets/c_cpp.snippets	2017-04-09 19:19:24.399924835 +0200
+++ c9fork/plugins/node_modules/ace/lib/ace/snippets/c_cpp.snippets	2017-04-10 10:44:41.500656371 +0200
@@ -129,3 +129,40 @@
 	[${1}](${2}){
 		${3:/* code */}
 	}${4}
+
+## Snippets by cyclone
+
+# if-else
+snippet ife
+	if (${1:true}) {
+		${2}
+	} else {
+		${0}
+	}
+	${3}
+
+snippet if
+	if (${1:true}) {
+		${2}
+	}
+	${3}
+
+snippet try
+	try {
+		${1:/* code */}
+	}
+	catch (${2:const char error[]}) {
+		${3:std::cout << error << std::endl;}
+	}
+	${4}
+
+snippet while
+	while (${1:true}) {
+		${2}
+	}
+
+# C version of for loop
+snippet forc
+	for (${2:i} = 0; $2 < ${1:count}; $2${3:++}) {
+		${4:/* code */}
+	}${5}
