diff -Naur c9fork/plugins/node_modules/architect-build/build_support/mini_require.js.orig c9fork/plugins/node_modules/architect-build/build_support/mini_require.js
--- c9fork/plugins/node_modules/architect-build/build_support/mini_require.js.orig      2018-08-20 12:27:03.980000000 +0200
+++ c9fork/plugins/node_modules/architect-build/build_support/mini_require.js   2018-08-20 12:27:08.020000000 +0200
@@ -474,11 +474,8 @@
     s.onerror = function(e) {
         console.log("Error: "+path);
         if (progress) progress.max = progress.max - 1;
-        processLoadQueue({
-            message: "Error loading script " + id + ":" + path,
-            id: id,
-            path: path
-        });
+        head.removeChild(s);
+        setTimeout(function() { loadScript(path, id, callback); }, 1000);
     };
     head.appendChild(s);
 };
