diff -Naur c9upstream/node_modules/architect-build/build_support/mini_require.js c9fork/node_modules/architect-build/build_support/mini_require.js
--- c9upstream/node_modules/architect-build/build_support/mini_require.js	2017-04-10 10:56:36.376844480 +0200
+++ c9fork/node_modules/architect-build/build_support/mini_require.js	2017-04-10 10:59:46.244805181 +0200
@@ -420,11 +420,8 @@
     s.onerror = function(e) {
         console.log("Error: "+path);
         if (progress) progress.max = progress.max - 1;
-        processLoadQueue({
-            message: "Error loading script " + id  + ":" + path,
-            id: id,
-            path: path
-        });
+        head.removeChild(s);
+        setTimeout(function() { loadScript(path, id, callback); }, 1000);
     };
     head.appendChild(s);
 };
