diff -Naur c9upstream/plugins/c9.ide.run/run.js.orig c9upstream/plugins/c9.ide.run/run.js
--- c9upstream/plugins/c9.ide.run/run.js.orig      2018-08-20 12:18:34.544000000 +0200
+++ c9upstream/plugins/c9.ide.run/run.js   2018-08-22 10:41:42.748000000 +0200
@@ -352,6 +352,9 @@
                 
                 cwd = insertVariables(cwd, options);
                 console.log(cmd);
+                
+                var username = c9.home.substr(c9.home.lastIndexOf("/")+1);
+                
                 // Execute run.sh
                 proc.tmux(cmd, {
                     session: procName,
@@ -363,7 +366,8 @@
                     rows: 5,
                     cwd: cwd,
                     validatePath: true,
-                    testing: testing
+                    testing: testing,
+                    tmuxName: username
                 }, function(err, pty, processId) {
                     if (err)
                         return callback(err);
@@ -627,10 +631,15 @@
                 
                 checking = [cb];
                 
+                var username = base.replace("/workspace/", "");
+                username = username.replace("/workspace", "");
+                username = username.substr(username.lastIndexOf("/")+1);
+                
                 // Execute run.sh
                 proc.tmux("", {
                     session: procName,
-                    fetchpid: true
+                    fetchpid: true,
+                    tmuxName: username
                 }, function(err, pty, newPid) {
                     var callbacks = checking || [];
                     checking = false;
diff -Naur c9fork/plugins/c9.ide.terminal/tmux_connection.js.orig c9fork/plugins/c9.ide.terminal/tmux_connection.js
--- c9fork/plugins/c9.ide.terminal/tmux_connection.js.orig      2018-08-20 12:05:24.732000000 +0200
+++ c9fork/plugins/c9.ide.terminal/tmux_connection.js   2018-08-20 12:04:58.652000000 +0200
@@ -128,13 +128,16 @@
         var cwd = session.cwd || "";
         if (!/^~(\/|$)/.test(cwd))
             cwd = session.root + cwd;
+        
+        var username = c9.home.substr(c9.home.lastIndexOf("/")+1);
         var command = "";
         var options = {
             cwd: cwd,
             cols: session.cols || 80,
             rows: session.rows || 24,
             name: "xterm-color",
-            base: installPath && installPath.replace(/^~/, c9.home || "~")
+            base: installPath && installPath.replace(/^~/, c9.home || "~"),
+            tmuxName: username
         };
         
         // Output Mode
@@ -305,9 +308,13 @@
         },
         kill: function(session) {
             console.warn("KILLING SESSION:", session.id);
+            var username = session.root.replace("/workspace/", "");
+            username = username.replace("/workspace", "");
+            username = username.substr(username.lastIndexOf("/")+1);
             
             if (session.id) {
                 proc.tmux("", {
+                    tmuxName: username,
                     session: session.id,
                     kill: true
                 }, function(err) {
